tests/transforms/test_split_non_commuting.py

```diff
@@ -122,765 +119,580 @@ def complex_wires_processing_fn(results):
-
-        with qml.queuing.AnnotatedQueue() as q:
-            obs_meas_1(op=qml.PauliX(0))
-            obs_meas_2(op=qml.PauliZ(1))
-            obs_meas_1(op=qml.PauliZ(0))
-            wire_meas_1(wires=[0])
-            wire_meas_2(wires=[1])
-            wire_meas_1(wires=[0, 1])
-
-        tape = qml.tape.QuantumScript.from_queue(q)
-        tapes, _ = split_non_commuting(tape)
-        assert len(tapes) == 2
-        assert tapes[0].measurements == [
-            obs_meas_1(op=qml.PauliX(0)),
-            obs_meas_2(op=qml.PauliZ(1)),
-            wire_meas_2(wires=[1]),
-        ]
-        assert tapes[1].measurements == [
-            obs_meas_1(op=qml.PauliZ(0)),
-            wire_meas_1(wires=[0]),
-            wire_meas_1(wires=[0, 1]),
-        ]
-
-    def test_grouping_strategies(self):
-        """Tests that the tape is split correctly for different grouping strategies"""
-
-        measurements = [
-            qml.expval(c * o) for c, o in zip([0.1, 0.2, 0.3, 0.4, 0.5], single_term_obs_list)
-        ]
-        tape = qml.tape.QuantumScript([], measurements, shots=100)
-
-        expected_tapes_no_grouping = [
-            qml.tape.QuantumScript([], [qml.expval(o)], shots=100) for o in single_term_obs_list
-        ]
-
-        # qwc grouping produces [[0, 3], [1, 2, 4]]
-        expected_tapes_qwc_grouping = [
-            qml.tape.QuantumScript([], [qml.expval(o) for o in group], shots=100)
-            for group in single_term_qwc_groups
-        ]
-
-        # wires grouping produces [[0, 2], [1], [3], [4]]
-        expected_tapes_wires_grouping = [
-            qml.tape.QuantumScript([], [qml.expval(o) for o in group], shots=100)
-            for group in single_term_wires_groups
-        ]
-
-        tapes, fn = split_non_commuting(tape, grouping_strategy=None)
-        for actual_tape, expected_tape in zip(tapes, expected_tapes_no_grouping):
-            qml.assert_equal(actual_tape, expected_tape)
-        assert qml.math.allclose(fn([0.1, 0.2, 0.3, 0.4, 0.5]), [0.01, 0.04, 0.09, 0.16, 0.25])
-
-        tapes, fn = split_non_commuting(tape, grouping_strategy="default")
-
-        for actual_tape, expected_tape in zip(tapes, expected_tapes_qwc_grouping):
-            qml.assert_equal(actual_tape, expected_tape)
-        assert qml.math.allclose(fn([[0.1, 0.2], [0.3, 0.4, 0.5]]), [0.01, 0.06, 0.12, 0.08, 0.25])
-
-        tapes, fn = split_non_commuting(tape, grouping_strategy="qwc")
-        for actual_tape, expected_tape in zip(tapes, expected_tapes_qwc_grouping):
-            qml.assert_equal(actual_tape, expected_tape)
-        assert qml.math.allclose(fn([[0.1, 0.2], [0.3, 0.4, 0.5]]), [0.01, 0.06, 0.12, 0.08, 0.25])
+    def test_single_hamiltonian_precomputed_grouping(
+        self, H, grouping_indices, results, expected_result
+    ):
+        """Tests that precomputed grouping of a single Hamiltonian is used."""
+
+        H._grouping_indices = grouping_indices  # pylint: disable=protected-access
```

Comment:
Is there an alternative way to access these values?

Reply:
This was resolved without comment. Same question as above.

Reply:
Sorry about that. I've updated the code to use the public `grouping_indices` instead, that's why I marked it as resolved.

Reply:
All good, thanks. Just making sure things don't go unnoticed :)